manager/allocator/networkallocator/networkallocator.go

```diff
@@ -85,6 +85,9 @@ type NetworkAllocator interface {
 
 	// IsAttachmentAllocated If lb endpoint is allocated on the node
 	IsAttachmentAllocated(node *api.Node, networkAttachment *api.NetworkAttachment) bool
+
+	// UpdateNetworkState updates the network state
+	UpdateNetworkState(network *api.Network) error
```

Comment:
Why do you need to add this method to two separate interfaces?

Reply:
Adding new methods to the `NetworkAllocator` interface is a breaking change: the CNMAllocator in github.com/moby/moby@v28.3.3 would not satisfy this interface despite satisfying the interface in github.com/moby/swarmkit/v2@v2.0.0. As I requested in my first review, please find a way to extend Swarmkit **without** introducing a breaking change that would necessitate bumping Swarmit's major version to 3.